users_ldap_groups/models/res_company_ldap.py

```diff
@@ -48,5 +55,16 @@ def _get_or_create_user(self, conf, login, ldap_entry):
                     "adding user %d to group %s", user, mapping.group_id.name,
                 )
                 groups.append((4, mapping.group_id.id, False))
+        if (
+            this.only_ldap_groups
+            and len([g[1] for g in groups if g[0] == 4 and g[1] in essential_groups])
+            != 1
+        ):
+            raise UserError(
+                _(
+                    "The created user needs to have one (and only one) of the"
```

Comment:
Why do you need this check? Odoo already assures this.

Reply:
Not exactly. This issue happens when we select the `only_ldap_groups` option, thus replacing all of the default groups added by Odoo by the ones we define. 
If we don't set one (and only one) of the user types, we will get an inconsistent behavior without a proper warning by Odoo, so IMHO it is better to give this error to the user and stop the creation of the new user.